features/cellular/framework/AT/AT_CellularDevice.cpp

```diff
@@ -230,6 +229,7 @@ nsapi_error_t AT_CellularDevice::get_sim_state(SimState &state)
             tr_error("SIM PUK required");
             break;
         case SimStateUnknown:
+            device_err_t err = _at->get_last_device_error();
```

Comment:
This might cause a race condition in multithread environment because not inside lock() - unlock() sequence

Reply:
Encapsulate in original place with #if MBED_CONF_MBED_TRACE_ENABLE - #endif

Reply:
I agree. I think it's better to read the error in the original place. It is also a bit more future proof than moving it into this specific switch case.